unixODBCDrivers.redshift: init at 1.4.49.1000

sir4ur0n opened the pull request:
- [x] Prerequisite: https://github.com/NixOS/nixpkgs/pull/168177 is merged

Add Amazon Redshift ODBC driver.

As Amazon does not provide sources, we extract the Debian package and manually patchelf.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
unixODBCDrivers.redshift: init at 1.4.49.1000

Add Amazon Redshift ODBC driver.

As Amazon does not provide sources, we extract the Debian package and manually patchelf.

sir4ur0n commented:
/marvin opt-in
/status needs_reviewer

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+
+  redshift = stdenv.mkDerivation rec {
+    pname = "redshift-odbc";
+    version = "1.4.45.1000-1";
```
```suggestion
    version = "1.4.45.1000";
```

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+    version = "1.4.45.1000-1";
+
+    src = fetchurl {
+      url = "https://s3.amazonaws.com/redshift-downloads/drivers/odbc/1.4.45.1000/AmazonRedshiftODBC-64-bit-${version}.x86_64.deb";
```
```suggestion
      url = "https://s3.amazonaws.com/redshift-downloads/drivers/odbc/${version}/AmazonRedshiftODBC-64-bit-${version}-1.x86_64.deb";
```

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+    installPhase = ''
+      mkdir -p $out/lib
+      cp opt/amazon/redshiftodbc/lib/64/* $out/lib
+      ${patchelf}/bin/patchelf --set-rpath ${unixODBC}/lib/ $out/lib/libamazonredshiftodbc64.so
```
```suggestion
      patchelf --set-rpath ${unixODBC}/lib/ $out/lib/libamazonredshiftodbc64.so
```
part of stdenv

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+    };
+
+    unpackPhase = ''
+      ${dpkg}/bin/dpkg -x $src src
```
```suggestion
      dpkg -x $src src
```
Please add it to nativeBuildInputs

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+
+    buildInputs = [ unixODBC ];
+
+    passthru =
+      {
+        fancyName = "Amazon Redshift (x64)";
+        driver = "lib/libamazonredshiftodbc64.so";
+      };
```
```suggestion
    passthru = {
      fancyName = "Amazon Redshift (x64)";
      driver = "lib/libamazonredshiftodbc64.so";
    };
```

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,45 @@
+      description = "Amazon Redshift ODBC driver";
+      homepage = "https://docs.aws.amazon.com/redshift/latest/mgmt/configure-odbc-connection.html";
+      license = licenses.unfree;
+      platforms = platforms.linux;
```
Missing maintainer

sir4ur0n commented:
Thank you @SuperSandro2000 for the thorough and qualitative review!

I'm a bit busy these days but I added this to my todolist, I'll fix all threads in the coming days hopefully!

sir4ur0n commented:
@SuperSandro2000 I think I addressed all your threads.

On top of that:
* I updated the version to 1.4.49.1000
* I updated the SHA accordingly
* I reworded the commit

Please review again :pray:

SuperSandro2000 commented on pkgs/development/libraries/unixODBCDrivers/default.nix:
```diff
@@ -174,4 +174,47 @@
+      sha256 = "sha256-r5HvsZjB7+x+ClxtWoONkE1/NAbz90NbHfzxC6tf7jA=";
+    };
+
+    nativeBuildInputs = [ dpkg patchelf ];
```
```suggestion
    nativeBuildInputs = [ dpkg ];
```
part of stdenv

sir4ur0n replied:
Done in the latest version
